e2e/update/src/update-mspgroups.e2e-spec.ts

```diff
@@ -0,0 +1,152 @@
+    });
+
+    it('05. should be able to remove multiple MSP Group numbers', () => {
+        const groupNum0 = groupData.groupNum = '1234567';
+        groupPage.navigateTo();
+        groupPage.clickButton('btn', 'Remove MSP Group');
+        groupPage.typeGroupNumber('Remove new MSP Group #1', 'goupNo_0', groupData);
+        const groupNum1 = groupData.groupNum = '7654321';
+        groupPage.clickButton('btn', 'Remove MSP Group');
+        groupPage.typeGroupNumber('Remove new MSP Group #2', 'goupNo_1', groupData);
+        groupPage.checkInputDisplayed('Remove new MSP Group #1', 'goupNo_0').then(function(val){
+            expect(val).toBe(groupNum0.toString());
+        });
+        groupPage.checkInputDisplayed('Remove new MSP Group #2', 'goupNo_1').then(function(val){
+            expect(val).toBe(groupNum1.toString());
+        });
+        groupPage.continue();
+        expect(browser.getCurrentUrl()).toContain(REVIEW_PAGE_URL, 'should navigate to the next page');
+    });
+
+    it('06. should be able to update multiple MSP Group numbers', () => {
+        const groupNum0 = groupData.groupNum = '1234567';
+        groupPage.navigateTo();
+        groupPage.clickButton('btn', 'Update administration of MSP Group');
+        groupPage.typeGroupNumber('Update administration of MSP Group #1', 'goupNo_0', groupData);
+        groupPage.checkThirdPartyAdmin('goupNo_0', 'Is the group', 'No');
+        const groupNum1 = groupData.groupNum = '7654321';
+        groupPage.clickButton('btn', 'Update administration of MSP Group');
+        groupPage.typeGroupNumber('Update administration of MSP Group #2', 'goupNo_1', groupData);
+        groupPage.checkThirdPartyAdmin('goupNo_1', 'Is the group', 'Yes');
+        groupPage.checkInputDisplayed('Update administration of MSP Group #1', 'goupNo_0').then(function(val){
+            expect(val).toBe(groupNum0.toString());
+        });
+        groupPage.checkInputDisplayed('Update administration of MSP Group #2', 'goupNo_1').then(function(val){
+            expect(val).toBe(groupNum1.toString());
+        });
+        groupPage.continue();
+        expect(browser.getCurrentUrl()).toContain(REVIEW_PAGE_URL, 'should navigate to the next page');
+    });
+
+    it('07. should be able to add, remove and\/or update at the same time', () => {
+        const groupNum0 = groupData.groupNum = '1234567';
+        groupPage.navigateTo();
+        groupPage.clickButton('btn', 'Add MSP Group');
+        groupPage.typeGroupNumber('Add new MSP Group #1', 'goupNo_0', groupData);
+        groupPage.checkThirdPartyAdmin('goupNo_0', 'Will this group', 'Yes');
+        const groupNum1 = groupData.groupNum = '7654321';
+        groupPage.clickButton('btn', 'Remove MSP Group');
+        groupPage.typeGroupNumber('Remove new MSP Group #1', 'goupNo_0', groupData);
+        groupPage.checkInputDisplayed('Add new MSP Group #1', 'goupNo_0').then(function(val){
```

Comment:
Don't worry about changing this one, but pro-tip for your own convenience: arrow functions. I pretty much never write anonymous functions anymore (`.then(function(){ ...})`.  Arrow functions are shorter and actually have a few small details which make them nicer.  So you'd  write:

```
groupPage.whatever().then(val => {
});
```